compiler_opt/tools/generate_default_trace.py

```diff
@@ -139,20 +138,19 @@ def main(_):
   config = registry.get_configuration()
 
   logging.info('Loading module specs from corpus.')
-  module_specs = corpus.build_modulespecs_from_datapath(
+  cps = corpus.Corpus(
       _DATA_PATH.value,
       additional_flags=config.flags_to_add(),
       delete_flags=config.flags_to_delete())
   logging.info('Done loading module specs from corpus.')
 
   if _MODULE_FILTER.value:
     m = re.compile(_MODULE_FILTER.value)
-    module_specs = [ms for ms in module_specs if m.match(ms.name)]
+    cps.filter(m)
 
   # Sampling if needed.
-  if _SAMPLING_RATE.value < 1:
-    sampled_modules = int(len(module_specs) * _SAMPLING_RATE.value)
-    module_specs = random.sample(module_specs, k=sampled_modules)
+  sampled_modules = int(len(cps) * _SAMPLING_RATE.value)
```

Comment:
add a unittest in corpus_test.py for the k == len(cps) case

Reply:
I had "test_bucket_sample_all", or did you mean to add another/something different?

Reply:
yeah, this is what i'm looking for, sorry i missed it before